app/assets/javascripts/discourse/app/components/d-modal.gjs

```diff
@@ -0,0 +1,357 @@
+import Component from "@glimmer/component";
+import { cached, tracked } from "@glimmer/tracking";
+import ClassicComponent from "@ember/component";
+import { concat } from "@ember/helper";
+import { on } from "@ember/modifier";
+import { action } from "@ember/object";
+import didInsert from "@ember/render-modifiers/modifiers/did-insert";
+import willDestroy from "@ember/render-modifiers/modifiers/will-destroy";
+import { cancel, later } from "@ember/runloop";
+import { service } from "@ember/service";
+import { modifier } from "ember-modifier";
+import { and, not, or } from "truth-helpers";
+import ConditionalInElement from "discourse/components/conditional-in-element";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import TrapTab from "discourse/modifiers/trap-tab";
+
+export const CLOSE_INITIATED_BY_BUTTON = "initiatedByCloseButton";
+export const CLOSE_INITIATED_BY_ESC = "initiatedByESC";
+export const CLOSE_INITIATED_BY_CLICK_OUTSIDE = "initiatedByClickOut";
+export const CLOSE_INITIATED_BY_SWIPE_DOWN = "initiatedBySwipeDown";
+export const CLOSE_INITIATED_BY_MODAL_SHOW = "initiatedByModalShow";
+
+const FLASH_TYPES = ["success", "error", "warning", "info"];
+const CLOSE_MODAL_DURATION = 250;
+const MIN_SWIPE_THRESHOLD = -10;
+const MAX_SWIPE_THRESHOLD = -150;
+
+export default class DModal extends Component {
+  @service modal;
+  @service site;
+
+  @tracked wrapperElement;
+  @tracked closing = false;
+
+  handleSwipeDown = modifier((element) => {
+    if (!this.site.mobileView) {
+      return;
+    }
+
+    element.addEventListener("touchstart", this.handleTouchStart, {
+      passive: true,
+    });
+    element.addEventListener("touchmove", this.handleTouchMove, {
+      passive: true,
+    });
+    element.addEventListener("touchend", this.handleTouchEnd, {
+      passive: true,
+    });
+
+    return () => {
+      element.removeEventListener("touchstart", this.handleTouchStart);
+      element.removeEventListener("touchmove", this.handleTouchMove);
+      element.removeEventListener("touchend", this.handleTouchEnd);
+    };
+  });
```

Comment:
Personally, I'm not a fan of these 'modifiers-with-access-to-class-internals'. To me it feels like the old classic-component didInsertElement hook, but with even more coupling and confusion 😓

this.backdrop = element; feels especially bad for coupling between the data layer and the presentation layer.

Defining a local modifier in this file, but outside the class definition would be nicer. Or even better, can we re-use discourse/modifiers/draggable? 👀

Reply:
> this.backdrop = element; feels especially bad for coupling between the data layer and the presentation layer.

I had already removed it, it was just a bit I had forgotten